apm-agent-plugins/apm-jms-plugin/src/test/java/co/elastic/apm/agent/jms/JmsInstrumentationIT.java

```diff
@@ -208,6 +219,25 @@ public void testQueueSendReceiveOnNonTracedThread() throws Exception {
         doTestSendReceiveOnNonTracedThread(() -> brokerFacade.receive(queue, 10), queue, false);
     }
 
+    @SuppressWarnings("ConstantConditions")
+    @Test
+    public void testQueueSendReceiveOnNonTracedThread_NonSampledTransaction() throws Exception {
+        final Queue queue = createTestQueue();
+
+        // End current transaction and start a non-sampled one
+        tracer.currentTransaction().deactivate().end();
+        reporter.reset();
```

Comment:
why do we need to reset the reporter here ?

Reply:
Because we `end` a transaction just above, which means it is reported, but soon after I use `reporter.getFirstTransaction()`, which I want to return my tested transaction and not this one